server/src/lib.rs

```diff
@@ -178,13 +180,15 @@ pub async fn main() -> std::io::Result<()> {
                     )
                     .service(
                         web::resource("/topic")
+                            .wrap(auth_middleware.clone())
                             .route(web::post().to(handlers::topic_handler::create_topic))
                             .route(web::delete().to(handlers::topic_handler::delete_topic))
                             .route(web::put().to(handlers::topic_handler::update_topic))
                             .route(web::get().to(handlers::topic_handler::get_all_topics)),
                     )
                     .service(
                         web::resource("/message")
+                            .wrap(auth_middleware.clone())
```

Comment:
Don't do this. In the Handler files, look at the params for the functions. The ones that require auth don't have an Option<LoggedUser>, but those that do just have LoggedUser. You need to write some code that can dynamically select between the two or edit the auth_handler code so that it has some way of switching between requiring auth and not.

Reply:
@densumesh https://github.com/arguflow/arguflow/pull/452/files#diff-f76d67463607cfe631fabd8a20e7e0d9aae0e2dd507d7a4ec561a6095da63cde is what I'm using here, isn't that what you're saying?